extension/app/src/pages/MainPage.tsx

```diff
@@ -1,40 +1,110 @@
 import {
   Button,
   ExternalLinkIcon,
+  LoginIcon,
   LogoHorizontalColorLogo,
-  MarkPenIcon,
+  LogoutIcon,
   Page,
-  TranslateIcon,
+  Spinner,
+  TextArea,
 } from "@dust-tt/sparkle";
+import { useEffect, useState } from "react";
 import { Link } from "react-router-dom";
+import { sendAuthMessage } from "../lib/auth";
 
 export const MainPage = () => {
+  const [token, setToken] = useState<string | null>(null);
+  const [isLoading, setIsLoading] = useState<boolean>(true);
+
+  useEffect(() => {
+    const storedToken = localStorage.getItem("authToken");
+    if (storedToken) {
+      setToken(storedToken);
+    }
+    setIsLoading(false);
+  }, []);
+
+  const handleLogin = () => {
+    setIsLoading(true);
+    sendAuthMessage()
+      .then((response) => {
+        console.log(response);
+        if (response?.accessToken) {
+          localStorage.setItem("authToken", response.accessToken);
```

Comment:
Can we encrypt the token before storing it. IIRC this is part of the best-practices around Chrome extension. Either you use the chrome identity API or encrypt the token before storing it.

Reply:
I used chrome storage api instead of local storage 👍🏻